src/main/java/org/ow2/proactive/catalog/callgraph/CatalogObjectCallGraphPDFGenerator.java

```diff
@@ -299,20 +320,33 @@ private void callGraphStyle(JGraphXAdapter<GraphNode, DefaultEdge> callGraphAdap
         graphComponent.setWheelScrollingEnabled(false);
     }
 
-    private CallGraphHolder buildCatalogCallGraph(List<CatalogObjectMetadata> catalogObjectMetadataList) {
+    private CallGraphHolder buildCatalogCallGraph(List<CatalogObjectMetadata> catalogObjectMetadataList,
+            CatalogObjectService catalogObjectService) {
 
         CallGraphHolder callGraphHolder = new CallGraphHolder();
         for (CatalogObjectMetadata catalogObjectMetadata : catalogObjectMetadataList) {
             List<String> dependsOnCatalogObjects = collectDependsOnCatalogObjects(catalogObjectMetadata);
             if (!dependsOnCatalogObjects.isEmpty()) {
-                GraphNode callingWorkflow = callGraphHolder.addNode(catalogObjectMetadata.getBucketName(),
-                                                                    catalogObjectMetadata.getName());
+                GraphNode callingCatalogObject = callGraphHolder.addNode(catalogObjectMetadata.getBucketName(),
+                                                                         catalogObjectMetadata.getName(),
+                                                                         catalogObjectMetadata.getKind(),
+                                                                         true);
 
                 for (String dependsOnCatalogObject : dependsOnCatalogObjects) {
                     String bucketName = separatorUtility.getSplitBySeparator(dependsOnCatalogObject).get(0);
-                    String workflowName = separatorUtility.getSplitBySeparator(dependsOnCatalogObject).get(1);
-                    GraphNode calledWorkflow = callGraphHolder.addNode(bucketName, workflowName);
-                    callGraphHolder.addDependsOnEdge(callingWorkflow, calledWorkflow);
+                    String objectName = separatorUtility.getSplitBySeparator(dependsOnCatalogObject).get(1);
```

Comment:
Are you verifying separatorUtility.getSplitBySeparator(dependsOnCatalogObject).length() > 1 ?

Reply:
dependsOnCatalogObject respect the following model: bucketName/ObjectName